### ISSUE TITLE
Hide unavailable bounties

### ISSUE THREAD

**Issue by rndquu** [devpool-directory/devpool-directory-tasks#11]
Check the [list](https://github.com/ubiquity/devpool-directory/issues) of all available partners' bounties.

When a bounty (i.e. issue) has the `Unavailable` label it means that the bounty is already assigned to some bounty hunter. We should hide such bounties from the [list](https://github.com/ubiquity/devpool-directory/issues).

What should be done:
- if bounty is unavailable (i.e. assigned to some bounty hunter) and does not exist in the list (i.e. first sync) then don't sync such bounty
- if bounty is unavailable (i.e. assigned to some bounty hunter) and exists in the list with the state `open` then close such bounty
- if bounty is available (i.e. not assigned to some bounty hunter) and exists in the list with the state `closed` then reopen such bounty

**Comment by ashmitjsg** [comment]
Hi @rndquu Can I work on this issue?

**Comment by rndquu** [comment]
> Hi @rndquu Can I work on this issue?

Hey, sure

**Comment by EtherealGlow** [comment]
How does devpool bounties sync all bounties, when does it know to run

**Comment by EtherealGlow** [comment]
@rndquu

**Comment by rndquu** [comment]
> How does devpool bounties sync all bounties, when does it know to run

[This](https://github.com/ubiquity/devpool-directory/blob/f1563a708695ed940e0eb2786a6818f3e206b343/.github/workflows/sync-issues.yml) github workflow syncs bounties (i.e. github issues) every 15 minutes

**Comment by BeanieMen** [comment]
a

**Comment by BeanieMen** [comment]
ill just make the pr first i guess

**Comment by BeanieMen** [comment]
[case 1](https://github.com/BeanieMen/devpool-directory/issues/1)
[case 2](https://github.com/BeanieMen/devpool-directory/issues/2)

**Comment by BeanieMen** [comment]
[case 3](https://github.com/BeanieMen/devpool-directory/issues/3)

**Comment by BeanieMen** [comment]
also this might be a really really shameless plug
but i have not received a review for 10 days in [here](https://github.com/ubiquity/.github/issues/90)

**Comment by 0x4007** [comment]
This isn't working @rndquu @Keyrxng 

This is currently assigned:

https://github.com/ubiquity/card-issuance/issues/26

![image](https://github.com/ubiquity/devpool-directory-bounties/assets/4975670/c16b6d15-6caa-41d5-a616-b9c3ab63005e)

Public and private issues are affected.

**Comment by rndquu** [comment]
> This isn't working @rndquu @Keyrxng
> 
> This is currently assigned:
> 
> [ubiquity/card-issuance#26](https://github.com/ubiquity/card-issuance/issues/26)
> 
> Public and private issues are affected.

https://github.com/ubiquity/devpool-directory-private doesn't have the latest code from https://github.com/ubiquity/devpool-directory

https://github.com/ubiquity/devpool-directory-private should have been forked from https://github.com/ubiquity/devpool-directory, @devpanther could you fix it?

**Comment by devpanther** [comment]
Oh @rndquu, I've forgotten but something did not work when we forked. 

Oh yeah, we can't private a fork of a public repo

**Comment by devpanther** [comment]
I don't have access to privates anymore

**Comment by rndquu** [comment]
> Oh @rndquu, I've forgotten but something did not work when we forked.
> 
> Oh yeah, we can't private a fork of a public repo

Did you try https://docs.github.com/en/repositories/creating-and-managing-repositories/duplicating-a-repository#mirroring-a-repository-in-another-location ?

**Comment by rndquu** [comment]
> I don't have access to privates anymore

https://github.com/ubiquity/work.ubq.fi/issues/2#issuecomment-2015254588

You need write access to https://github.com/ubiquity/devpool-directory-private, right?

**Comment by devpanther** [comment]
> > I don't have access to privates anymore
> 
> https://github.com/ubiquity/work.ubq.fi/issues/2#issuecomment-2015254588
> 
> You need write access to https://github.com/ubiquity/devpool-directory-private, right?

Yes

**Comment by rndquu** [comment]
> > > I don't have access to privates anymore
> > 
> > 
> > [ubiquity/work.ubq.fi#2 (comment)](https://github.com/ubiquity/work.ubq.fi/issues/2#issuecomment-2015254588)
> > You need write access to https://github.com/ubiquity/devpool-directory-private, right?
> 
> Yes

Sent invite with write access to https://github.com/ubiquity/devpool-directory-private

**Comment by 0x4007** [comment]
This needs to be addressed in time for the conference. @Keyrxng can you handle this?

Otherwise @ubiquity/software-development any takers?

**Comment by devpanther** [comment]
> This needs to be addressed in time for the conference. @ubiquity/software-development  any takers?

On it already, turning the private to a mirror and that's it

**Comment by molecula451** [comment]
what is the fuzz with this task, why it was reopened, it had 2 linked PRs assuming solving this issue?

**Comment by 0x4007** [comment]
> if bounty is unavailable (i.e. assigned to some bounty hunter) and exists in the list with the state open then close such bounty

This is not implemented correctly. I receive frequent complaints from new would-be candidates that its tedious to go through the [devpool.directory](https://devpool.directory/) and constantly have to sift through already assigned tasks. 

This is especially relevant as we are hiring a team of developer relations managers to find devs at the upcoming conference this week, with their first point of entry being the [devpool.directory](https://devpool.directory/)

**Comment by rndquu** [comment]
> > if bounty is unavailable (i.e. assigned to some bounty hunter) and exists in the list with the state open then close such bounty
> 
> This is not implemented correctly. I receive frequent complaints from new would-be candidates that its tedious to go through the [devpool.directory](https://devpool.directory/) and constantly have to sift through already assigned tasks.
> 
> This is especially relevant as we are hiring a team of developer relations managers to find devs at the upcoming conference this week, with their first point of entry being the [devpool.directory](https://devpool.directory/)

The is implemented correctly. As you already [mentioned](https://github.com/ubiquity/devpool-directory-bounties/issues/11#issuecomment-2014687367) you see already assigned issues only from private repos. 

TLDR;
1. https://github.com/ubiquity/devpool-directory-private doesn't have the latest code, we're fixing it [here](https://github.com/ubiquity/work.ubq.fi/issues/2#issuecomment-2015254588)
2. This issue can be closed as completed

**Comment by 0x4007** [comment]
> The is implemented correctly. As you already mentioned you see already assigned issues only from private repos.

No my comment asserts that public and private repos are affected. Besides, I just checked again and the first task from a public repos is assigned. 

![image](https://github.com/ubiquity/devpool-directory-bounties/assets/4975670/d7372732-61ce-41ae-8794-94054f0dc613)

![image](https://github.com/ubiquity/devpool-directory-bounties/assets/4975670/3b37544c-2062-4135-8462-61311f422dc6)

**Comment by Keyrxng** [comment]
# Open in UBQ devpool:

![empty_string_unassigned](https://github.com/ubiquity/devpool-directory-bounties/assets/106303466/86dac172-bfb8-4729-834d-9fad9aca1fae)


<hr/>

# Open in Keyrxng devpool:

![empty_string_k_devpool](https://github.com/ubiquity/devpool-directory-bounties/assets/106303466/fc15c1b0-41bd-4d6f-a47b-5441f42bfe62)


<hr/>

# I assigned myself it making it unavailable:

![devpool_assigned](https://github.com/ubiquity/devpool-directory-bounties/assets/106303466/7cebbd31-de39-4675-aae9-e3866ec71ad7)

<hr/>

# Closed in keyrxng devpool because it's now assigned 
![devpool_k_closed_because_assigned](https://github.com/ubiquity/devpool-directory-bounties/assets/106303466/ec0274bb-552c-4ea7-8882-b099ef0fb4bb)

**Comment by Keyrxng** [comment]
- https://github.com/ubiquity/devpool-directory/pull/1145

My current open PR against the devpool is the logic which ran to produce this result. However, the original issue was resolved according to the spec here and that was just to remove the `unavailable` bounties from the devpool repo was it not? The work UI interfacing was not part of this bounty

**Comment by 0x4007** [comment]
> However, the original issue was resolved according to the spec here and that was just to remove the `unavailable` bounties from the devpool repo was it not? 

You're right. I read the following as "closing" the issue which means that it would not render in the work.ubq.fi UI

> if bounty is unavailable (i.e. assigned to some bounty hunter) and exists in the list with the state open then close such bounty

With an emphasis on 

> then close such bounty

So we need to file a new issue and get a new pull request to actually close them?

**Comment by Keyrxng** [comment]
> > However, the original issue was resolved according to the spec here and that was just to remove the `unavailable` bounties from the devpool repo was it not?
> 
> You're right. I read the following as "closing" the issue which means that it would not render in the work.ubq.fi UI
> 
> > if bounty is unavailable (i.e. assigned to some bounty hunter) and exists in the list with the state open then close such bounty
> 
> With an emphasis on
> 
> > then close such bounty
> 
> So we need to file a new issue and get a new pull request to actually close them?

- https://github.com/ubiquity/devpool-directory/issues/1121

Sorry confused that PR with the decimals one but once the next action run happens im certain that decimals will be closed,

Although the QA I presented and if you check the devpool, it states this issue is closed and you have just left a comment on the PR associated with it

**Comment by rndquu** [comment]
FYI current issue (hiding unavailable bounties) works as expected in https://github.com/ubiquity/devpool-directory/pull/1145

**Comment by 0x4007** [comment]
> FYI current issue (hiding unavailable bounties) works as expected in https://github.com/ubiquity/devpool-directory/pull/1145

I see, thanks for highlighting this. 

The proper form of delivery would require cherry picking changes and associating them with a task to be eligible for the reward.

**Comment by Keyrxng** [comment]
- https://github.com/Keyrxng/devpool-directory/actions/runs/8433798301
- https://github.com/Keyrxng/devpool-directory/issues?q=is%3Aopen+is%3Aissue+label%3AUnavailable
- https://github.com/Keyrxng/devpool-directory/actions/runs/8433848461 (manually run to activate updates)

Hiding the `unavailable` bounties was implemented correctly when rndquu closed this as completed. I just deleted all issues and then checked-out from the merge commit with the only change being commenting out the twitter stuff and no `unavailable` labels or assigned issues exist in my devpool
